### DISCUSSION TITLE
v0.10.1: Unstable Release

### DISCUSSION THREAD

**Discussion by Amazinite** [endless-sky/endless-sky#8809] [Announcements]
This is an unstable release, containing big changes that may introduce new bugs. Major new changes include the following:

Content and features:
* Released the first chapter of Hai Reveal, a storyline that involves the open secret of the Hai being exposed to humanity.
* NPCs in missions are now able to trigger game actions on status changes to the NPC.
* Updated and added new mechanics around asteroid mining to make it a more engaging system, including allowing your escorts to take a more active role in mining.
* Continued development of the Gegno, including new ships and outfits.
* Split the Quarg government into a series of regional governments, allowing for more fine-tuned interactions with different groups of Quarg.
* Plugins are now able to be enabled and disabled within the Plugins menu.
* Replaced the mouse control toggle button with a preference to prevent accidental activations.
* Controls can now be unbound by pressing X or delete when hovering over the control.

Bug fixes:
* Patched an infinite money glitch caused by bad offer conditions on an Unfettered mission.
* Ships with low max speeds no longer rubberband backwards when exiting hyperspace.
* Fixed various issues with buying and selling outfits caused by having outfits in cargo or storage.

For a complete list of changes, as well as further detail about the summarized changes, see the [changelog](https://github.com/endless-sky/endless-sky/blob/master/changelog). Special thanks to those who contributed to this release: @10010101001, @1010todd, @3agleEmpire, @alextd, @Amazinite, @Anarchist2, @Arachi-Lover, @AvianGeneticist, @Azure3141, @bene-dictator, @DeBlister, @Deltaspace0, @dorbarker, @EjoThims, @Ferociousfeind, @Fzzr, @Hurleveur, @Koranir, @LorenzoBolla, @MasterOfGrey, @mOctave, @petervdmeer, @Quantumshark, @quyykk, @RisingLeaf, @rovermicrover, @samrocketman, @Saugia, @shitwolfymakes, @Terin, @thewierdnut, @tibetiroka, @TomGoodIdea, @UnorderedSigh, @warp-core, @ziproot, @Zitchas

If you are using the high-dpi plugin, you should download and install the [updated version](https://github.com/endless-sky/endless-sky-high-dpi/releases/tag/v0.10.1) that goes with this release.

![image](https://github.com/endless-sky/endless-sky/assets/17688683/26f1b216-047f-4b24-93c5-4fbc5878979a)

<hr /><em>This discussion was created from the release <a href='https://github.com/endless-sky/endless-sky/releases/tag/v0.10.1'>v0.10.1: Unstable Release</a>.</em>

**Comment by lumbar527** [comment]
> A note for all players: in v0.10.1, we changed the way licenses and dominated planets are stored in the save file, so loading a save file from v0.10.1 or newer in an older version of the game will erase all licenses and dominated planets. It is still safe to load a save file from an older version of the game in newer versions.

How did it change?

**Reply by warp-core** [reply]
They were previously saved as conditions but now are saved in separate structures with on demand autoconditions available to access or update them in game.

**Reply by lumbar527** [reply]
Sorry, I'm not entirely sure what that means. What do they look like (the "code")?

**Reply by warp-core** [reply]
They used to be stored like this:
```
conditions
	"license: Coalition"
	"license: Pilot's"
	"license: Remnant"
	"license: Remnant Capital"
```

Now, they're stored like this:
```
licenses
	Coalition
	Pilot's
	Remnant
	"Remnant Capital"
```

**Reply by lumbar527** [reply]
Can you set them the same way in missions? Also, what about tribute? (Thanks for showing me this, ~I hope I don't break my save file~)
Edit: Turns out my save file is up-to-date and I don't have to break something!

**Reply by warp-core** [reply]
Yes, missions can still interact with them the same way as before.
Tributes were previously stored as something like:
```
conditions
	"tribute: <planet>" <credits>
```
Now it is:
```
"tribute received"
	<planet> <credits>
```

**Reply by lumbar527** [reply]
Thanks! (Why isn't there a wiki page for save-file reading?)

**Reply by warp-core** [reply]
Because no one has made one.
I believe there is a very old guide on the Steam community page somewhere, but of course, it won't have these recent changes.
Here: https://steamcommunity.com/sharedfiles/filedetails/?id=723485559&searchtext=how+to+save

**Reply by lumbar527** [reply]
I'm currently searching for requests for one and will probably make an issue about this. That could be a related issue.
